Java/core/src/main/java/com/amazon/randomcutforest/RandomCutForest.java

```diff
@@ -154,9 +169,21 @@
      */
     protected final boolean storeSequenceIndexesEnabled;
     /**
-     * Store the time information
+     * Enable compact representation
      */
     protected final boolean compactEnabled;
+    /**
+     * Enable float representation; the input/output is double[] but the internal
+     * representation would be float[]. There will be loss of precision in near
+     * neighbors because of the truncation. Otherwise, unless all numbers are in a
+     * very small range, the loss of precision is unlikely to have larger impact
+     * than changing the random seed.
+     */
+    protected final boolean floatEnabled;
```

Comment:
Instead of a boolean, what if we defined an enum for precision?
```
public enum Precision {
  SINGLE, DOUBLE;
}

public static final DEFAULT_PRECISION = Precision.DOUBLE;

protected final Precision precision;
```

Reply:
Name changed to singlePrecisionEnabled. Agree with the possibility of enum. All of these flags could do with some rethinking in a future PR.

Reply:
https://github.com/aws/random-cut-forest-by-aws/issues/77